src/urllib3/_collections.py

```diff
@@ -90,6 +91,13 @@ def __init__(
         self._container = OrderedDict()
         self.lock = RLock()
 
+        warnings.warn(
+            "RecentlyUsedContainer is deprecated and scheduled for removal in urllib3.future v3. "
```

Comment:
Whenever possible, Try to keep the interfacing classes and just map the new functionality on them.